src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -169,43 +169,58 @@ private void processZipFileIfEligible(CloudBlobContainer container, String zipFi
             logAbortedProcessingNonExistingFile(zipFilename, container.getName());
         } else {
             cloudBlockBlob.downloadAttributes();
-            processZipFile(container, cloudBlockBlob, zipFilename);
+            leaseAndProcessZipFile(container, cloudBlockBlob, zipFilename);
         }
     }
 
-    private void processZipFile(
+    private void leaseAndProcessZipFile(
         CloudBlobContainer container,
         CloudBlockBlob cloudBlockBlob,
         String zipFilename
     ) throws StorageException, IOException {
-        Optional<String> leaseId = blobManager.acquireLease(cloudBlockBlob, container.getName(), zipFilename);
+        Optional<String> leaseIdOption = blobManager.acquireLease(cloudBlockBlob, container.getName(), zipFilename);
 
-        if (leaseId.isPresent()) {
-            Envelope envelope = envelopeProcessor.getEnvelopeByFileAndContainer(container.getName(), zipFilename);
+        if (leaseIdOption.isPresent()) {
+            String leaseId = leaseIdOption.get();
 
-            if (envelope == null) {
-                // Zip file will include metadata.json and collection of pdf documents
-                try (ZipInputStream zis = loadIntoMemory(cloudBlockBlob, zipFilename)) {
-                    createEvent(ZIPFILE_PROCESSING_STARTED, container.getName(), zipFilename, null);
+            try {
+                processZipFile(container, cloudBlockBlob, zipFilename, leaseId);
+            } finally {
+                blobManager.tryReleaseLease(cloudBlockBlob, container.getName(), zipFilename, leaseId);
+            }
+        }
+    }
 
-                    ZipFileProcessingResult processingResult =
-                        processZipFileContent(zis, zipFilename, container.getName(), leaseId.get());
+    private void processZipFile(
```

Comment:
Diff is terrible, but this method is extracted logic from the one above, so that it's separate from lease management.